index.js

```diff
@@ -1,101 +1,136 @@
 'use strict'
-const _ = require('lodash')
 const co = require('co')
-const Condition = require('@ripple/five-bells-condition').Condition
-const UnmetConditionError = require('@ripple/five-bells-shared/errors/unmet-condition-error')
-const TransferAPI = require('./lib/transfer-api')
-const Subscriber = require('./lib/subscriber')
-const utils = require('./lib/utils')
+const formats = require('./lib/formats/optimistic')
+const EventEmitter = require('events')
+const RippleAPI = require('ripple-lib').RippleAPI
+
+const find = require('lodash/fp/find')
+const findXrpBalance = find(['currency', 'XRP'])
 
 // options -
-//   ledger_id   - String, e.g. "wss://s.altnet.rippletest.net:51233"
-//   credentials - {address, secret}
+//   server      - String, e.g. "wss://s.altnet.rippletest.net:51233"
+//   address     - String, Ripple address
+//   secret      - String, Ripple secret
 //   log         - A logger instance
 //   sourceSubscriptions
-function RippledLedger (options) {
-  this.id = options.ledger_id
-  this.credentials = options.credentials // {address, secret}
-  this.log = options.log || console
-  this.sourceSubscriptions = options.sourceSubscriptions
-  this.client = new TransferAPI(this.id, this.credentials, this.log)
-
-  this.client.transferPool.on('expire', function (transfers) {
-    co(this._rejectTransfers.bind(this), transfers)
-      .catch(this.onExpireError.bind(this))
-  }.bind(this))
-}
+class RipplePlugin extends EventEmitter {
+  constructor (options) {
+    super()
 
-RippledLedger.TYPE = 'https://ripple.com/ilp/v1'
-RippledLedger.validateTransfer = require('./lib/validate')
+    this.server = options.server
+    this.address = options.address
```

Comment:
Options shouldn't include the prefix. But when calling `getAccount` it'll be included.